robotpy_build/hooks.py

```diff
@@ -67,7 +69,10 @@ class Hooks:
     _qualname_trans = str.maketrans(_qualname_bad, "_" * len(_qualname_bad))
 
     def __init__(
-        self, data: HooksDataYaml, casters: typing.Dict[str, str], report_only: bool
+        self,
+        data: HooksDataYaml,
+        casters: typing.Dict[str, typing.Dict[str, typing.Any]],
```

Comment:
Hm, why do we have a `dict[str, Any]` instead of a `TypedDict` or something else?